consensus/core/src/core.rs

```diff
@@ -351,10 +371,22 @@ impl Core {
         // There must be a quorum of blocks from the previous round.
         let quorum_round = self.threshold_clock.get_round().saturating_sub(1);
 
+        let leader_authority = &self
+            .context
+            .committee
+            .authority(self.first_leader(quorum_round))
+            .hostname;
+
         // Create a new block either because we want to "forcefully" propose a block due to a leader timeout,
         // or because we are actually ready to produce the block (leader exists and min delay has passed).
         if !force {
             if !self.leaders_exist(quorum_round) {
+                self.context
+                    .metrics
+                    .node_metrics
+                    .block_proposal_leader_wait_count
```

Comment:
I think we should use a separate metric for counting the number of times leader is not found. `block_proposal_leader_wait_count` is tied to `block_proposal_leader_wait_ms`, so when the average wait is ~250ms, we know the leader is missing.

Reply:
I think the confusion for me with these metrics is that it doesn't just include leader wait time, it includes the quorum receive wait time which can make this metric a little misleading. Separating them brings more clarity. Though I guess we could always subtract this metric from quorum receive latency.

Reply:
Looking at `ThresholdClock::add_block()`, `quorum_ts` is updated when a quorum forms. Then the `block_proposal_leader_wait_ms` is calculated from `quorum_ts`. This does not include quorum receive latency. Also, leader timeout task is signaled when `quorum_ts` is updated, so `block_proposal_leader_wait_ms` should line up with actual leader wait.

Reply:
What I was trying to measure is the specific moment when the leader block is received versus when the round can advance due to quorum received. The wait time is anytime after the quorum is received as that is the minimum needed to advance the round. The leader block can be received and make up the quorum needed to advance the clock. So to accurately get the wait time for a leader block we should get the `min(leader_received_ts - quorum_ts, 0)`. 

What we currently have in theory should lead to the same results but after we add the block we go through the process of `try_commit` which then adds to the wait time calculation for the leader. I think this is why we see the difference in the graphs. (pasting here again)

[left is the new metric and right is main]
![image](https://github.com/user-attachments/assets/16959d94-a241-4f9f-a887-99e470b5dc96)

Reply:
Is the average is calculated with block_proposal_leader_wait_ms / block_proposal_leader_wait_count, can you compare the `block_proposal_leader_wait_count` rate as well? Originally the intention is to have the count similar to # of blocks proposed. With the new logic, I think it is double counting: if a block proposal has to wait for 250ms, and `try_new_block()` gets called after waiting for 50ms, 100ms, 150ms and 200ms, then the average leader wait seems to be skewed higher.

Reply:
Yeah you are correct using leader count was incorrect in the new metric but comparing it looks like the wait is indeed lower still

https://metrics.sui.io/goto/mZanqCiHg?orgId=1